Sources/PortalSwift/Components/PortalWebView.swift

```diff
@@ -177,29 +219,24 @@ public class PortalWebView: UIViewController, WKNavigationDelegate, WKScriptMess
     self.onPageStart?()
   }
 
-  /// Called when the web view finishes loading a page. It injects the Portal script into the web view.
+  /// Called when the web view finishes loading a page.
   /// - Parameters:
   ///   - webView: The WKWebView instance that finished loading.
   ///   - navigation: The navigation information associated with the event.
   public func webView(_: WKWebView, didFinish _: WKNavigation!) {
-    guard let address = portal.address else {
-      print("[PortalWebView] No address found for user. Cannot inject provider into web page.")
-      return
-    }
-
-    let javascript = self.injectPortal(
-      address: address,
-      apiKey: self.portal.apiKey,
-      chainId: String(self.portal.chainId),
-      gatewayConfig: self.portal.gatewayConfig[self.portal.chainId]!,
-      autoApprove: self.portal.autoApprove,
-      enableMpc: true
-    )
-
-    self.evaluateJavascript(javascript, sourceURL: "portal_sign")
-
     self.onPageComplete?()
```

Comment:
Did you move the code above to a function called onPageComplete? or how is the script getting injected now?

Reply:
i see you moved it to the init function for the injectPortal function call

Reply:
The injection script is moved to a WKUserScript to be loaded via the WKWebView configuration with the injectionTime: .atDocumentStart property set to guarantee our script loads first.